src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -1,32 +1,90 @@
 import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.transaction.Resource;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
 import uk.gov.companieshouse.overseasentitiesapi.mapper.OverseasEntityDtoDaoMapper;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionCreatedResponseDto;
 import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
 import uk.gov.companieshouse.overseasentitiesapi.repository.OverseasEntitySubmissionsRepository;
 import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
 
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.FILING_KIND_OVERSEAS_ENTITY;
+
 @Service
 public class OverseasEntitiesService {
 
     private final OverseasEntitySubmissionsRepository overseasEntitySubmissionsRepository;
+    private final TransactionService transactionService;
 
     private final OverseasEntityDtoDaoMapper overseasEntityDtoDaoMapper;
 
     @Autowired
     public OverseasEntitiesService(OverseasEntitySubmissionsRepository overseasEntitySubmissionsRepository,
+                                   TransactionService transactionService,
                                    OverseasEntityDtoDaoMapper overseasEntityDtoDaoMapper) {
         this.overseasEntitySubmissionsRepository = overseasEntitySubmissionsRepository;
+        this.transactionService = transactionService;
         this.overseasEntityDtoDaoMapper = overseasEntityDtoDaoMapper;
     }
 
-    public void createOverseasEntity(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
-        ApiLogger.debug("Called createOverseasEntity()");
+    public ResponseEntity<Object> createOverseasEntity(Transaction transaction,
+                                                       OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+                                                       String passthroughTokenHeader) throws ServiceException {
+        ApiLogger.debug("Called createOverseasEntity(...)");
+
+        if (hasExistingOverseasEntitySubmission(transaction)) {
+            return ResponseEntity.badRequest().body(String.format("Transaction id: %s has an existing Overseas Entity submission", transaction.getId()));
+        }
 
+        // add the overseas entity submission into MongoDB
         var overseasEntitySubmissionDao = overseasEntityDtoDaoMapper.dtoToDao(overseasEntitySubmissionDto);
-        var insertedSubmissionDao = overseasEntitySubmissionsRepository.insert(overseasEntitySubmissionDao);
+        var insertedSubmission = overseasEntitySubmissionsRepository.insert(overseasEntitySubmissionDao);
+        var submissionUri = String.format("/transactions/%s/overseas-entity/%s", transaction.getId(), insertedSubmission.getId());
```

Comment:
I think it would be good to define this submission  URI and the validation status URI below as constants somewhere. Perhaps privately in this class, or if we have more callback URIs going on elsewhere, maybe a new string constants class. Just to collect them together - would give an overview of the URIs.